File: sanha/sanha/report/sm_report/sm_report.js
// sm_report.js

frappe.query_reports["SM Report"] = {
    "filters": [
        {
            "fieldname": "query_types",
            "label": __("Query Types"),
            "fieldtype": "Link",
            "options": "Query Types",
            "default": "",
        },
    ],

    onload: function (report) {
        // Customization of the header section to add the logo and client information
        const header = $(report.page.main.find(".page-header-content"));
        header.empty().append(`
            <div style="text-align: center;">
                <img src="/files/your_logo.png" alt="Company Logo" style="width: 150px;"/>
                <h3>Client Information</h3>
                <div>Client Name: ${frappe.user.full_name}</div>
                <div>Client Code: ${frappe.user.user_code}</div>
            </div>
        `);
    },

    formatter: function (value, row, column, data, default_formatter) {
        // This handles specific row customizations if needed.
        if (column.fieldname === "query_status" && data.query_status !== "Draft") {
            value = `<span style="color: green;">${value}</span>`;
        }
        return default_formatter(value, row, column, data);
    }
};

File: sanha/sanha/report/sm_report/sm_report.py
# sm_report.py

import frappe
from frappe import _

def execute(filters=None):
    columns, data = get_columns(), get_data(filters)
    return columns, data

def get_columns():
    return [
        {"label": _("Serial No"), "fieldname": "serial_no", "fieldtype": "Data", "width": 50},
        {"label": _("Raw Material"), "fieldname": "raw_material", "fieldtype": "Data", "width": 150},
        {"label": _("Supplier"), "fieldname": "supplier", "fieldtype": "Data", "width": 150},
        {"label": _("Manufacturer"), "fieldname": "manufacturer", "fieldtype": "Data", "width": 150},
        {"label": _("Query Status"), "fieldname": "workflow_state", "fieldtype": "Data", "width": 150},
    ]

def get_data(filters):
    query = """
        SELECT 
            ROW_NUMBER() OVER() as serial_no,
            raw_material,
            supplier,
            manufacturer,
            workflow_state AS query_status
        FROM
            `tabQuery Doctype`
        WHERE 
            workflow_state != 'Draft'
        ORDER BY 
            raw_material ASC
    """
    data = frappe.db.sql(query, as_dict=True)
    return data
